File: Data Structures & Algorithms/merge-intervals/submission-1.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> res;
        sort(intervals.begin(), intervals.end());
        for(const auto &interval: intervals){
            if(res.empty()|| res.back()[1]<interval[0]){
                res.push_back(interval);
            }
            else{
                res.back()[1]= max(res.back()[1], interval[1]);
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-1.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public int minMeetingRooms(List<Interval> intervals) {
        Collections.sort(intervals, (a, b)->Integer.compare(a.start, b.start));

        PriorityQueue<Integer> minheap= new PriorityQueue<>();
         for(Interval i: intervals){
            if(!minheap.isEmpty() && minheap.peek()<=i.start){
                minheap.poll();
            }
            minheap.offer(i.end);
         }
         return minheap.size();

    }
}
